lib/linter.js

```diff
@@ -688,6 +688,22 @@ function markVariableAsUsed(scopeManager, currentNode, parserOptions, name) {
     return false;
 }
 
+/**
+ * Gets all the ancestors of a given node
+ * @param {ASTNode} node The node
+ * @returns {ASTNode[]} All the ancestor nodes in the AST, not including the provided node, starting
+ * from the root node and going inwards to the parent node.
+ */
+function getAncestors(node) {
+    if (node.parent) {
+        const parentAncestors = getAncestors(node.parent);
```

Comment:
Does this need to be recursive? I would love an iterative approach here, if that is feasible.